org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/codeactions/LSPCodeActionsMenu.java

```diff
@@ -113,31 +113,15 @@ public IStatus runInUIThread(IProgressMonitor monitor) {
 						} else if (t != null) {
 							for (Either<Command, CodeAction> command : t) {
 								if (command != null) {
-									if (command.isLeft()) {
-										final MenuItem item = new MenuItem(menu, SWT.NONE, index);
-										item.setText(command.getLeft().getTitle());
-										item.addSelectionListener(new SelectionAdapter() {
-											@Override
-											public void widgetSelected(SelectionEvent e) {
-												executeCommand(info, command.getLeft());
-											}
-										});
-									} else if (command.isRight()) {
-										CodeAction codeAction = command.getRight();
-										final MenuItem item = new MenuItem(menu, SWT.NONE, index);
-										item.setText(codeAction.getTitle());
-										item.addSelectionListener(new SelectionAdapter() {
-											@Override
-											public void widgetSelected(SelectionEvent e) {
-												if (codeAction.getEdit() != null) {
-													LSPEclipseUtils.applyWorkspaceEdit(codeAction.getEdit(), codeAction.getTitle());
-												}
-												if (codeAction.getCommand() != null) {
-													executeCommand(info, codeAction.getCommand());
-												}
-											}
-										});
-									}
```

Comment:
I do not understand why we can change this code as part of the PR. What if the server does not support resolveCodeAction?

Reply:
> I do not understand why we can change this code as part of the PR

It's just some better factorization.

>  What if the server does not support resolveCodeAction?

I believe this case is already handled in CodeActionCompletionProposal, so the factorization fixes it.